File: Plugins/Aspose_Tasks_Java_for_Ruby/lib/asposetasksjava/Projects/readinggroupdefinitiondata.rb
module Asposetasksjava
  module ReadingGroupDefinitionData
    def initialize()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'
                
        # Instantiate project object
        project = Rjb::import('com.aspose.tasks.Project').new(data_dir + 'test_tasks.mpp')

        puts "Task Groups Count: " + project.getTaskGroups().size().to_s

        task_group = project.getTaskGroups().toList().get(0)
        puts "Percent Complete:" +  task_group.getName().to_s
        puts "Group Criteria count: " + task_group.getGroupCriteria().size().to_s
    end
  end
end

File: Plugins/Aspose_Tasks_Java_for_PHP/src/aspose/tasks/WorkingWithProjects/ReadingGroupDefinitionData.php
<?php
namespace Aspose\Tasks\WorkingWithProjects;

use com\aspose\tasks\Project as Project;
class ReadingGroupDefinitionData{

    public static function run($dataDir=null){

        # Instantiate project object
        $project = new Project($dataDir . 'test_tasks.mpp');

        print "Task Groups Count: " . $project->getTaskGroups()->size().PHP_EOL;//.to_s;

        $task_group = $project->getTaskGroups()->toList()->get(0);
        print "Percent Complete:" .  (string)$task_group->getName().PHP_EOL;//.to_s
        print "Group Criteria count: " . (string)$task_group->getGroupCriteria()->size().PHP_EOL;//.to_s
    }

}
?>